sdk/analytics/src/test/java/com/klaviyo/analytics/state/SideEffectTests.kt

```diff
@@ -209,25 +215,102 @@ class SideEffectTests : BaseTest() {
+        Registry.register<State>(klaviyoStateMock)
+        StateSideEffects(
+            state = klaviyoStateMock,
+            apiClient = apiClientMock
+        )
+
+        capturedApiObserver.captured(
+            mockk<ProfileApiRequest>().apply {
+                every { status } returns KlaviyoApiRequest.Status.Failed
+                every { responseCode } returns 400
+                every { responseBody } returns """
+                    {
+                      "errors": [
+                        {
+                          "id": "67ed6dbf-1653-499b-a11d-30310aa01ff7",
+                          "status": 400,
+                          "code": "invalid",
+                          "title": "Invalid input.",
+                          "detail": "Invalid phone number format (Example of a valid format: +12345678901)",
+                          "source": {
+                            "pointer": "/data/attributes/phone_number"
+                          },
+                          "links": {},
+                          "meta": {}
+                        }
+                      ]
+                    }
+                """.trimIndent()
+            }
+        )
+
+        verify { klaviyoStateMock.resetPhoneNumber() }
+        Registry.unregister<State>()
+    }
+
+    @Test
+    fun `Invalid input on email resets field`() {
+        Registry.register<State>(klaviyoStateMock)
+        StateSideEffects(
+            state = klaviyoStateMock,
+            apiClient = apiClientMock
+        )
+
+        capturedApiObserver.captured(
+            mockk<ProfileApiRequest>().apply {
+                every { status } returns KlaviyoApiRequest.Status.Failed
+                every { responseCode } returns 400
+                every { responseBody } returns """
+                    {
+                      "errors": [
+                        {
+                          "id": "4f739784-390b-4df3-acd8-6eb07d60e6b4",
+                          "status": 400,
+                          "code": "invalid",
+                          "title": "Invalid input.",
+                          "detail": "Invalid email address",
+                          "source": {
+                            "pointer": "/data/attributes/email"
+                          },
+                          "links": {},
+                          "meta": {}
+                        }
+                      ]
+                    }
+                """.trimIndent()
+            }
+        )
+
+        verify { klaviyoStateMock.resetEmail() }
+        Registry.unregister<State>()
+    }
+
     @Test
     fun `Other API failures do not affect push state`() {
         StateSideEffects(stateMock, apiClientMock)
 
         capturedApiObserver.captured(
             mockk<ProfileApiRequest>().apply {
                 every { status } returns KlaviyoApiRequest.Status.Failed
+                every { responseCode } returns 412
```

Comment:
Since we now check for response code in the `afterApiRequest` logic flow, I had to add this